search/ranker.hpp

```diff
@@ -101,7 +87,7 @@ class Ranker
   void FlushViewportResults(Geocoder::Params const & geocoderParams, Results & res);
 
   void SetParams(Params const & params) { m_params = params; }
-
+  void SetPreResults1(vector<PreResult1> * preResults1) { m_preResults1 = preResults1; }
```

Comment:
You can completely get rid of pointers - just pass vector by rvalue reference and take the ownership here.